### PR TITLE
Don't move a file under a directory when renaming

### PR DESCRIPTION
**Pull request by jelly**
We rename in Files by moving the file to the new filename but when this is a directory `mv` moves the file to that directory instead. This is quite logical in a shell but not in a GUI, there it is unexpected behaviour.

Closes: #399

### CODE REVIEW

File: src/fileActions.jsx [hunk 1]
```diff
@@ -206,8 +206,8 @@ const RenameItemModal = ({ path, selected }) => {
             : path.join("/") + "/" + name;
 
         cockpit.spawn(is_current_dir
-            ? ["mv", path.join("/"), newPath]
-            : ["mv", path.join("/") + "/" + selected.name, newPath],
+            ? ["mv", "-T", path.join("/"), newPath]
```
Comment (cockpituous) [inline src/fileActions.jsx:209]: This added line is not executed by any test.

Comment (jelly) [reply src/fileActions.jsx:209]: Hmm this should be covered. Let's try to cover that in a follow up but let's think about it:

This is about renaming the current dir, which we no longer support, so this is dead code.

Comment (jelly) [reply src/fileActions.jsx:209]: https://github.com/cockpit-project/cockpit-files/issues/405

**Review by allisonkarlitskaya — DISMISSED** [review]
Looks good, thanks!

I guess we get something like "targetname: Is a directory"?

Tests would be nice ;)

**Comment by jelly** [comment]
> Looks good, thanks!
> 
> I guess we get something like "targetname: Is a directory"?

We get this: https://github.com/cockpit-project/cockpit-files/pull/404/files#diff-e86755442bc7d21f0414014b3c356b54d6650e8ed9bffcc4f5858d1154df80e1R556

> Tests would be nice ;)

I've added a test, see the `test/check-application` changes.

**Comment by jelly** [comment]
Turns, out we likely want just `--no-clobber`, let's also not write short arguments but write them out so the code is readable.

**Comment by jelly** [comment]
This is flaky or coreutils changed behaviour.

**Comment by jelly** [comment]
Yup, this is broken. Needs work.

**Comment by jelly** [comment]
```
[root@archlinux admin]# mv -T  newfile dest && echo 1
mv: cannot overwrite directory 'dest' with non-directory 'newfile'
[root@archlinux admin]# mv -T -n  newfile dest && echo 1
1
```
Yeah this is great :) It is likely a bug in coreutils as it happens in rawhide as well. Fedora 40 has 9.4 while Arch has 9.5

**Comment by allisonkarlitskaya** [comment]
> As alternative @allisonkarlitskaya suggested passing `-i` instead, hoping that this will work with `cockpit.spawn` and that we can pass `y` or `n` via `.input()` on stderr. This sounds like some nice async work!

If you don't want to do the async work, there's a somewhat easier version:

```
$ cpf spawn -- mv -i -T a b stderr=message : done ch1 : wait | spy -m cockpit.bridge
...
{
  "exit-status": 1,
  "message": "mv: overwrite 'b'? ",
  "command": "close",
  "channel": "ch1"
}
```

with the successful case looking like

```
$ cpf spawn -- mv -i -T a c stderr=message : done ch1 : wait | spy -m cockpit.bridge
...
{
  "exit-status": 0,
  "message": "",
  "command": "close",
  "channel": "ch1"
}
```

The key here is `done`.  If you don't send it, the channel will hang waiting for the answer.  If you do send it, then `mv` will even give you a nice non-zero exit code.

**Comment by jelly** [comment]
Le JS code:
```
        cockpit.spawn([
            "mv", "--interactive", "--no-target-directory",
            path.join("/") + "/" + selected.name, newPath
        ],
                      { superuser: "try" }).input()
                .then(() => Dialogs.close())
                .catch(err => {
                    setNameError(err.message);
                });
```

**Comment by jelly** [comment]
This needs to be expedited as renaming a folder into a new folder still works which is wrong.

**Comment by jelly** [comment]
Superseeded by https://github.com/cockpit-project/cockpit-files/pull/459

**Pull request closed without merging**